web/packages/teleterm/src/mainProcess/mainProcessClient.ts

```diff
@@ -265,3 +263,32 @@ export default function createMainProcessClient(): MainProcessClient {
     },
   };
 }
+
+/** Receives and confirms messages sent through `AwaitableSender`.*/
+function makeAwaitableReceiver<T>(listener: (value: T) => void): {
+  port: MessagePort;
+  close: () => void;
+} {
+  // The order of ports doesn't matter.
+  const { port1, port2 } = new MessageChannel();
+
+  port1.onmessage = (event: MessageEvent<Message>) => {
+    const msg = event.data;
+    if (msg.type !== 'data') {
+      return;
+    }
+    listener(msg.payload as T);
```

Comment:
> There's no guarantee that the renderer has actually received and processed the message.

Then this `listener` here has to be sync, otherwise the renderer can be still technically processing the message even after it sends an ack to the main process, right?

Reply:
Currently we expect the listener to be sync, but it can be easily changed to support asynchronous listeners.